erp-server/src/main/java/com/ttt/erp/service/AwardPdfService.java

```diff
@@ -0,0 +1,129 @@
+
+    @Autowired
+    private EmailService emailService;
+
+
+    // class-level constants
+    private static final String OUTPUT_DIR = "awards/";
+    private static final PdfNumber LANDSCAPE = new PdfNumber(90);
+
+
+    private static ByteArrayInputStream generateAwardPdf(Award award) throws IOException {
+        // gather the components needed to build the file name
+        Employee employee = award.getEmployee();
+        UserAccount userAccount = award.getUserAccount();
+        AwardType awardType = award.getAwardType();
+
+        // setup the pdf document
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        PdfWriter writer = new PdfWriter(output, new WriterProperties().setPdfVersion(PdfVersion.PDF_2_0));
+        PdfDocument pdfDocument = new PdfDocument(writer);
+        pdfDocument.setTagged();
+        PageSize pageSize = PageSize.LETTER.rotate();
+        pdfDocument.setDefaultPageSize(pageSize);
+        Document document = new Document(pdfDocument);
+
+        // add the TTT logo
+        String logoSrc = "src/main/resources/logo.png";
+        Image logo = new Image(ImageDataFactory.create(logoSrc)).scaleAbsolute(200, 50);
+        document.add(logo);
+
+        document.add(new Paragraph(awardType.getName()));
+        document.add(new Paragraph("awarded to"));
+        document.add(new Paragraph(employee.getFullName()));
+
+        if (award.getDescription() != null) {
+            document.add(new Paragraph(award.getDescription()));
+        }
+
+        if (award.getAwardedDate() != null) {
+            document.add(new Paragraph("Awarded on "));
+            document.add(new Paragraph(award.getAwardedDate().toString()));
+        }
+
+        if (award.getAwardedTime() != null) {
+            document.add(new Paragraph(" at "));
+            document.add(new Paragraph(award.getAwardedTime().toString()));
+        }
+
+        document.add(new Paragraph(" by "));
+        document.add(new Paragraph(userAccount.getUsername()));
+
+        if (userAccount.getSignature() != null) {
+            String signatureSrc = "signatures/" + userAccount.getSignature();
+            Image signature = new Image(ImageDataFactory.create(signatureSrc)).scaleAbsolute(100, 100);
+            document.add(signature);
+
+//            PdfFileSpec fileSpec = PdfFileSpec.createEmbeddedFileSpec(pdfDocument, signatureSrc, "signature.png", PdfName.ApplicationOctetStream);
+        }
+
+        document.close();
+        return new ByteArrayInputStream(output.toByteArray());
+    }
+
+
+    public ResponseEntity<Long> sendAwardPdf(Long id) throws FileNotFoundException {
+        try {
+            // gather the components needed to build the file name
+            Award award = repository.findById(id);
+            Employee employee = award.getEmployee();
+
+            String filename = employee.getFullName() + " - Award.pdf";
+            String subjectLine = "Congratulations on award from TTT";
```

Comment:
Only comment here is that may be good to say "...from <user who created award>" in the subject and message. That's a total nitpick though, this all looks really good.

Reply:
Good idea. I'll make the change.